packages/react-ui/src/app/features/ai/ai-assistant-conversation.tsx

```diff
@@ -0,0 +1,58 @@
+import { MessageType } from '@/app/features/ai/lib/types';
+import { UseChatHelpers } from '@ai-sdk/react';
+import {
+  AIChatMessage,
+  AIChatMessageRole,
+  AIChatMessages,
+  LoadingSpinner,
+  MarkdownCodeVariations,
+} from '@openops/components/ui';
+import { useMemo } from 'react';
+
+type AiAssistantConversationnProps = {
+  isPending: boolean;
+  messages: MessageType[];
+} & Pick<UseChatHelpers, 'status'>;
+
+const ChatStatus = {
+  STREAMING: 'streaming',
+  SUBMITTED: 'submitted',
+};
+
+const AiAssistantConversation = ({
+  messages,
+  status,
+  isPending,
+}: AiAssistantConversationnProps) => {
+  const uiMessages: AIChatMessage[] = useMemo(() => {
+    return messages.map((message: MessageType, idx) => ({
+      id: message && 'id' in message ? message.id : String(idx),
+      role:
+        message.role.toLowerCase() === 'user'
+          ? AIChatMessageRole.user
+          : AIChatMessageRole.assistant,
```

Comment:
need to check, but we might need to filter tool messages once tools are also enabled